File: Basic Programming/Dimik oj/Python3/প্রোগ্রামিং সমস্যা 22 - [৫২ সমস্যা বই] মৌলিক সংখ্যা.py
#                      Name : Arif Khan
#                      Judge: Dimik Oj
#                      University: Primeasia University
#                      problem: প্রোগ্রামিং সমস্যা 22 - [৫২ সমস্যা বই] মৌলিক সংখ্যা
#                      Difficulty: Easy
#                      Time Complexity: 0.4388 সেকেন্ড
#                      Problem Link: https://dimikoj.com/problems/22/prime-number-by
#


import math

if __name__ == '__main__':
	prime = [True]*100001
	prime[0] = False
	prime[1] = False
	
	for i in range(2, math.ceil(math.sqrt(100001))):
		if prime[i] == True:
			j = 2
			while i * j < 100001:
				prime[i * j] = False
				j += 1
				
	for _ in range(int(input())):
		a, b = map(int,input().split())
		ans = 0
		for i in range(a,b+1,1):
			if prime[i] == True:
				ans += 1
		print(ans)

File: Basic Programming/Dimik oj/C++/প্রোগ্রামিং সমস্যা 22 - [৫২ সমস্যা বই] মৌলিক সংখ্যা.cpp
/*                    Name : Arif Khan
                      Judge: Dimik Oj
                      University: Primeasia University
                      problem: প্রোগ্রামিং সমস্যা 22 - [৫২ সমস্যা বই] মৌলিক সংখ্যা
                      Difficulty: Easy
                      Time Complexity: 0.2823 সেকেন্ড
                      Problem Link: https://dimikoj.com/problems/22/prime-number-by
*/

#include<bits/stdc++.h>
using namespace std;

bool prime[100001];

void primeCheck()
{
	memset(prime, true,sizeof(prime));
	prime[1] = false;prime[0] = false;
	for(int i = 2; i <= sqrt(100001) ; i++) {
		if(prime[i]) {
			for(int j = 2; i*j <= 100001; j++){
				prime[i*j] = false;
			}
		}
	}
}

int main()
{
    int t, a, b;
    scanf("%d",&t);
    primeCheck();
    for(int j = 0; j < t; j++) {
    	cin >> a >> b;
    	int ans = 0;
    	for(int i = a; i <= b; i++) {
    		if(prime[i]) ans++;
    	}
    	cout << ans << endl;
    }
    return 0;
}